internal/pkg/vm/actor/builtin/initactor/init.go

```diff
@@ -232,14 +247,14 @@ func (a *Impl) Exec(rt invocationContext, codeCID cid.Cid, params []interface{})
 
 		// map id to address and vice versa
 		ctx := context.TODO()
-		state.AddressMap, err = a.setId(ctx, rt.Runtime().Storage(), state.AddressMap, actorAddr, actorID)
+		state.AddressMap, err = setId(ctx, rt.Runtime().Storage(), state.AddressMap, actorAddr, actorID)
```

Comment:
In the case of account actors, I don't think this is right. For account actors, we need to map the actor's pubkey address (i.e. params[0]) to the ID address. The `actorAddr` should not be used.

See https://github.com/filecoin-project/specs/issues/762 though before diving in to fix it now.

Reply:
@anorth I made this work by having `rt.CreateActor` return the public key address in the case where the actor is an Account actor. There wasn't much guidance in the spec at the time. You're right though that nothing would work if this mapping didn't map to the public key address.